torch/utils/collect_env.py

```diff
@@ -243,6 +247,56 @@ def get_nvidia_smi():
     return smi
 
 
+def get_intel_gpu_onboard(run_lambda):
+    lst = []
+    platform = get_platform()
+    if platform == "linux":
+        txt = run_and_read_all(run_lambda, "xpu-smi discovery -j")
+        if txt:
+            try:
+                obj = json.loads(txt)
+                if type(obj["device_list"]) is list:
+                    for o in obj["device_list"]:
+                        lst.append(f'* {o["device_name"]}')
+                else:
+                    lst.append("N/A")
+            except (ValueError, TypeError) as e:
+                lst.append(str(e))
+        else:
+            lst.append("N/A")
+    if platform == "win32" or platform == "cygwin":
+        txt = run_and_read_all(
+            run_lambda,
+            'powershell.exe "gwmi -Class Win32_PnpSignedDriver | where{$_.DeviceClass -eq \\"DISPLAY\\"\
+            -and $_.Manufacturer -match \\"Intel\\"} | Select-Object -Property DeviceName | ConvertTo-Json"',
+        )
+        try:
```

Comment:
Do we need to check txt is not empty before try `json.load` just like linux do.

Reply:
done